insomniac/session.py

```diff
@@ -217,6 +223,9 @@ def run(self):
                 print("(use " + COLOR_BOLD + "--no-speed-check" + COLOR_ENDC + " to skip this check)")
                 sleeper.update_random_sleep_range()
 
+            if args.speed is not None:
```

Comment:
```suggestion
            if args.speed is not None:
                sleeper.set_random_sleep_range(args.speed)
            elif not args.no_speed_check:
                print("Checking your Internet speed to adjust the script speed, please wait for a minute...")
                print("(use " + COLOR_BOLD + "--no-speed-check" + COLOR_ENDC + " to skip this check)")
                sleeper.update_random_sleep_range()
```

Reply:
Much cleaner and no need to the check in params.py

Reply:
I see what you mean, I'll update the PR.